packages/@aws-cdk/aws-efs/lib/filesystem.ts

```diff
@@ -0,0 +1,137 @@
+import { Construct } from '@aws-cdk/cdk';
+import { cloudformation} from './efs.generated';
+
+/**
+ * Properties of a reference to an Elastic File System.
+ *
+ * @see FileSystemRef#import
+ * @see FileSystemRef#export
+ */
+export interface FileSystemRefProps {
+  /**
+   * The physical, human-readable Id of the EFS we're referencing.
+   * The EFS must be in the same account and region as the root Stack.
+   */
+  fileSystemId: string;
+}
+
+/* tslint:disable:max-line-length */
+export interface FileSystemProps {
+  /**
+   * The throughput, measured in MiB/s, that you want to provision for a file system that
+   * you're creating. The limit on throughput is 1024 MiB/s. You can get these limits
+   * increased by contacting AWS Support. Must be greater than or equal to 1.0MiB/s
+   *
+   * https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-efs-filesystem.html#cfn-elasticfilesystem-filesystem-provisionedthroughputinmibps
+   * @default 100
+   */
+  throughputMiB?: number;
+
+  /**
+   * The throughput mode for the file system to be created. There are two throughput modes to
+   * choose from for your file system: bursting and provisioned. You can decrease your file
+   * system's throughput in Provisioned Throughput mode or change between the throughput modes
+   * as long as it’s been more than 24 hours since the last decrease or throughput mode change.
+   *
+   * https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-efs-filesystem.html#cfn-elasticfilesystem-filesystem-throughputmode
+   * @default ThroughputMode.Bursting
+   */
+  throughputMode?: ThroughputMode
+
+  /**
+   * The PerformanceMode of the file system. We recommend generalPurpose performance mode for
+   * most file systems. File systems using the maxIO performance mode can scale to higher levels
+   * of aggregate throughput and operations per second with a tradeoff of slightly higher
+   * latencies for most file operations. This can't be changed after the file system has
+   * been created.
+   *
+   * https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-efs-filesystem.html#cfn-efs-filesystem-performancemode
+   * @default PerformanceMode.GeneralPurpose
+   */
+  performanceMode?: PerformanceMode
+
+  /**
+   * A boolean value that, if true, creates an encrypted file system. For more information, see
+   * CreateFileSystem in the Amazon Elastic File System User Guide.
+   *
+   * https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-efs-filesystem.html#cfn-efs-filesystem-encrypted
+   * @default true
+   */
+  encrypted?: boolean
+}
+/* tslint:enable:max-line-length */
+
+export abstract class FileSystemRef extends Construct {
+  public static import(parent: Construct, id: string, props: FileSystemRefProps) {
```

Comment:
Make sure to explicitly specify the return type here `FileSystemRef` - otherwise jsii will be angry that you are returning an unexported type